data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -129,7 +129,7 @@ public FileAppender<Record> newAppender(OutputFile outputFile, FileFormat fileFo
   public FileAppender<Record> newAppender(
       EncryptedOutputFile encryptedOutputFile, FileFormat fileFormat) {
     MetricsConfig metricsConfig =
-        table != null ? MetricsConfig.forTable(table) : MetricsConfig.fromProperties(config);
+        table != null ? MetricsConfig.forTable(table) : MetricsConfig.getDefault();
```

Comment:
How sure we about this change?
This could cause behavioral change as the `config` is not considered anymore.

Reply:
If we accept that the `config` is not considered anymore, then we can double check the `validateMetricsConfig` logic and simplify it too. We went to extra length to only do validation if the `table` is not set.

Maybe first we just throw an exception if the `write.metadata.metrics.` is set in the `config`?